plugins/core/src/lib/nodes/events/messageEvent.ts

```diff
@@ -18,6 +19,15 @@ const messageEventConfig = {
     write('event', event)
     write('content', event.content)
     commit('flow')
+
+    saveGraphEvent({
+      sender: event.sender,
+      agentId: event.agentId,
+      connector: 'onMessage',
+      connectorData: JSON.stringify(event.data),
+      content: event.content,
+      eventType: 'Message',
+    })
```

Comment:
I think we should add this into the BasePlugin class, where the event is processed by the plugin. Right in here:

```typescript
 mapEventsToEventBus() {
    this.events.forEach(event => {
      this.eventEmitter.on(event.eventName, async payload => {
        payload.eventName = event.eventName
        this.logger.debug(
          `Sending event ${event.eventName} to central message bus`
        )
        this.centralEventBus.emit(this.eventQueueName, payload)
        // await this.eventQueue.addJob(event.eventName, payload)
      })
    })
  }
  ```
  
  I also think we might want to store much of the event in a JSONB format, especially since we can query by JSONB properties.  The current event shape looks like the follow.  We don't need all of it, as some of it is mostly for runtime, but some may be important, especially since not all events coming in will have their data just on "content" (I don't think):
  
  ```typescript
  export type EventPayload<
  T = Record<string, unknown>,
  Y = Record<string, unknown>
> = {
  connector: string
  eventName: string
  status: 'success' | 'error' | 'pending' | 'unknown'
  content: string
  sender: string
  observer: string
  client: string
  channel: string
  plugin: string
  agentId: string
  isPlaytest?: boolean
  spellId?: string
  // entities: any[]
  channelType: string
  rawData: unknown
  timestamp: string
  stateKey?: string
  runInfo?: {
    spellId: string
  }
  data: T
  metadata: Y
}
```

Reply:
Maybe a good mid ground is to keep all your hard coded event properties, and then add a JSONB field for the full event.

We should also consider embedding the event contents automatically so they can always be queried by a user. We can do that in a follow up feature push.